Use experimental web as default

inpercima opened the pull request:
This PR moves the experimental web to used web interface for lvz-viz.

> There was a need to add the .gitattributes for *.bat files, otherwise the following gradle check will fail.
> As a result, the gradlew.bat was also checked in again.
> 
> ```bash
> > Task :editorconfigCheck FAILED
> gradlew.bat@1,5: Insert cr - violates end_of_line = crlf, reported by org.ec4j.linters.TextLinter
> gradlew.bat@2,52: Insert cr - violates end_of_line = crlf, reported by org.ec4j.linters.TextLinter
> ```

The problem with gradlew.bat deosn't exists anymore.

inpercima commented:
Hey @sepe81 maybe you have some little time to check this PR in small steps? Maybe you have an idea about the cahnges in gradlew.bat and what do you think about the basic idea of how it is done in this PR with the frontend/backend split?

inpercima commented:
The most changes here are the move from root to backend and the removal of the now unused "old" frontend.

sepe81 reviewed (changes requested):
@inpercima Could you please rebase with main and resolve the conflicts so we can continue with local testing?

inpercima commented:
This PR is from May, therefore I will rewrite it and adapt some things to current conditions.

> May we keep the backend in the root directory without an extra subfolder to simplify the restructuring and the local Java build?

Is the problem an IDE dependency? Your IDE or some other plugins/extensions search java for the root?

Then I will change that.
The idea was to just separate the Frontend and Backend so they could be developed separately.
If that causes problems in different scenarios, then we'll do it the way you suggested for now and see about separating them later.

inpercima committed:
chore(): add .gitattributes file

inpercima committed:
chore(): use resources, words and links from original app, integrate and style them in new app

inpercima committed:
refactor(): move frontend code to new location

inpercima committed:
refactor(): remove now unused old frontend code

inpercima committed:
refactor(): update frontend to get data from backend

inpercima commented on .github/workflows/java_ci.yml:
```diff
@@ -2,14 +2,14 @@ name: Java CI with Gradle
 on:
   push:
-    branches: [master]
+    branches: [use-experimental-web-as-default]
```
I had already changed the behavior earlier, so this is no longer necessary.

inpercima commented on frontend/README.md:
```diff
@@ -0,0 +1,100 @@
+
+Applicationwide title of the app, displayed in title and toolbar.
+
+* default: `LVZ Polizeiticker`
```
Changed.

inpercima commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,89 @@
+
+on:
+  push:
+    branches: [use-experimental-web-as-default]
```
I had already changed the behavior earlier, so this is no longer necessary.

inpercima committed:
build(): merge backend and frontend in the build process

inpercima committed:
fix(docs): correct spelling errors in prerequisites section of README

inpercima committed:
refactor(build): update Dockerfile and build.gradle for improved frontend build process

sepe81 committed:
fix(docs): correct grammar and consistency in README

sepe81 committed:
fix(build): correct buildFrontend task to run both install and build

The Gradle Exec task only executes the last commandLine declaration.
Changed to use shell command chaining to run both pnpm install and
pnpm build:prod sequentially.

sepe81 committed:
fix(search): prevent memory leak in paginator subscription

Added takeUntilDestroyed to the paginator.page subscription to ensure
it is cleaned up when the component is destroyed.

sepe81 committed:
fix(search): make distinctUntilChanged compare input values

distinctUntilChanged was comparing event objects which are always
different. Now maps to the actual input value before comparing,
preventing unnecessary API calls when the value hasn't changed.

sepe81 committed:
refactor(build): replace deprecated $buildDir with layout.buildDirectory

The $buildDir property is deprecated in newer Gradle versions.
Use the recommended layout.buildDirectory API instead.

sepe81 committed:
fix(search): improve search UX with Enter key and fix duplicate calls

- Add Enter key handler via template binding (keyup.enter) for immediate
  search - more reliable with Angular Material than fromEvent
- Fix duplicate API calls by setting paginator.pageIndex directly
  instead of calling firstPage() which could emit an additional event
- Pressing Enter always triggers search, even with same term

inpercima commented on build.gradle:
```diff
@@ -107,3 +107,29 @@ test {
+task buildFrontend(type: Exec) {
+  description = 'Builds the frontend.'
+  workingDir 'frontend'
+  commandLine 'pnpm', 'install --frozen-lockfile'
```
Thanks for your tests and comments. I will update the line.

inpercima commented on frontend/src/app/features/search/search.html:
```diff
@@ -14,8 +14,8 @@
           <input matInput placeholder="Suchphrase eingeben" #input />
```
I know what you mean. I just tested it again, and now something is wrong with the events. Possibly due to the latest Angular update, but I'm not sure. I'll definitely take a look at it.

sepe81 committed:
refactor(build): split buildFrontend into install and build tasks

This removes the dependency on 'sh -c' and improves cross-platform
compatibility by using separate Gradle tasks for installation and
building of the frontend.

inpercima committed:
feat(search): refactor search component to use reactive signals

inpercima committed:
fix(docs): downgrade default version to 2.5.0

inpercima committed:
feat(build): add GitHub Actions workflow for automatic version bumping

inpercima commented on frontend/src/app/features/statistic/statistic.ts:
```diff
@@ -32,7 +30,7 @@ import { StatisticServiceMock } from './statistic.service.mock';
   ],
 })
-export class Statistic implements OnInit, AfterViewInit {
+export class Statistic implements AfterViewInit {
```
What did you mean with main? Does it now work for you on this feature branch?
For me it works:
<img width="878" height="853" alt="image" src="https://github.com/user-attachments/assets/ba77101a-7700-48b0-82b3-89aa70b79e32" />

inpercima replied:
This seems to work for me too.

It would be interesting to know why it works for me in different browsers, even before the new commit. I'll look into that again.
Nevertheless, the commit is good, it makes the use of leaflet clearer.

inpercima committed:
fix(workflow): add trigger for version bump on push to main branch

sepe81 committed:
fix(statistic): load leaflet.heat via dynamic import for esbuild compatibility

The esbuild-based Angular builder does not expose L globally, causing
leaflet.heat's heatLayer to be undefined. Use a dynamic import after
assigning L to window so the plugin can find it at evaluation time.

sepe81 committed:
fix(statistic): ensure leaflet.heat initialization on extensible global L

The esbuild-based Angular builder imports 'leaflet' as a read-only ES
module namespace object. When this object was assigned directly to
window.L, the leaflet.heat plugin failed to register its 'heatLayer'
method because the object was non-extensible, leading to a
ReferenceError: L is not defined.

Changes:
- Use Object.assign({}, L) to provide an extensible copy of Leaflet
  to the global window object.
- Access 'heatLayer' explicitly from window.L after the dynamic import.
- Wrap map initialization in a try-catch block within ngAfterViewInit
  for better error visibility.
- Add a defensive guard in addToMap to prevent crashes if content
  loads before the heat layer is ready.

inpercima committed:
feat(release): implement new release workflow and rename version bump to release

inpercima commented:
Hey @sepe81, just a little reminder here for Q1 ;)

In my opinion, it's done; we can merge it for now and fix any bugs as soon as possible.
Should I remove the workflows for now and we'll look at it later if this is a blocker for you?

sepe81 reviewed (approved):
@inpercima Alright, let’s go – merging now!
